src/vm.rs

```diff
@@ -102,8 +101,8 @@ pub(crate) enum VmExecutionResult {
 // structure & storage, and tracks the read set of the current execution.
 // TODO: Simplify this type, like grouping [from] and [to] into a
 // [preprocessed_addresses] or a [preprocessed_locations] vector.
-struct VmDb<'a, S: Storage> {
-    vm: &'a Vm<'a, S>,
+struct VmDb<'a, S: Storage, C: PevmChain> {
```

Comment:
Looking back, propagating a lot of `PevmChain` here seems overkill. We can just pass in a `chain_id` instead?